proc.go

```diff
@@ -265,3 +265,37 @@ func (p Proc) fileDescriptors() ([]string, error) {
 func (p Proc) path(pa ...string) string {
 	return p.fs.Path(append([]string{strconv.Itoa(p.PID)}, pa...)...)
 }
+
+// FileDescriptorsInfo retrieves information about all file descriptors of
+// the process.
+func (p Proc) FileDescriptorsInfo() ([]ProcFDInfo, error) {
+	names, err := p.fileDescriptors()
+	if err != nil {
+		return nil, err
+	}
+
+	fdinfos := make([]ProcFDInfo, len(names))
+
+	for i, n := range names {
+		fdinfo, _ := p.FDInfo(n)
+		fdinfos[i] = *fdinfo
+	}
+
+	return fdinfos, nil
+}
+
+// InotifyWatchLen returns the total number of inotify watches used
+// by the process.
+func (p Proc) InotifyWatchLen() (int, error) {
```

Comment:
It might be better to make this a regular function which takes a slice of `ProcFDInfo` as an argument, instead of a method of Proc.  Since the process is not used other than to get the FDInfos.  Or you could define a new type which is a `[]ProcFDInfo` and make this a method of that type.